src/main/java/codeu/model/store/persistence/PersistentDataStore.java

```diff
@@ -149,6 +149,41 @@ public PersistentDataStore() {
     return messages;
   }
 
+  /**
+   * Loads Activity objects from the Datastore service that were created before the provided datetime,
+   * up to the provided limit of number of entries and sorted in descending order by creation time.
+   *
+   * @param startDatetime The retrieved activities would have been created exclusively before the startDatetime
+   * @param limit         The max. number of entries to retrieve
+   * @return A list of activities that met the criteria and sorted by creation time
+   * @throws PersistentDataStoreException if an error was detected during the load from the
+   *                                      Datastore service
+   */
+  public List<Activity> loadActivitiesBeforeDatetime(Instant startDatetime, int limit) throws PersistentDataStoreException {
+    Query query = new Query("chat-activities")
+            .setFilter(new FilterPredicate("datetime", FilterOperator.LESS_THAN, startDatetime.toString()))
+            .addSort("datetime", SortDirection.DESCENDING);
+
+    List<Entity> results = datastore.prepare(query).asList(FetchOptions.Builder.withLimit(limit));
+    List<Activity> activities = new ArrayList<>();
+
+    for (Entity entity : results) {
+      try {
+        UUID id = UUID.fromString((String) entity.getProperty("id"));
+        String description = (String) entity.getProperty("description");
+        Instant datetime = Instant.parse((String) entity.getProperty("datetime"));
+        int type = Math.toIntExact((long) entity.getProperty("type"));
```

Comment:
By default the numeric integer retrieved from the entity is a `long`, but I would find that wasteful for a type field, that certainly would not exceed the hundreds order of magnitude